quipucords/api/aggregate_report/model.py

```diff
@@ -37,62 +39,90 @@
+        default=0, blank=True, null=True
+    )
+    instances_hypervisor = models.PositiveIntegerField(default=0, blank=True, null=True)
+    instances_not_redhat = models.PositiveIntegerField(default=0, blank=True, null=True)
+    instances_physical = models.PositiveIntegerField(default=0, blank=True, null=True)
+    instances_unknown = models.PositiveIntegerField(default=0, blank=True, null=True)
+    instances_virtual = models.PositiveIntegerField(default=0, blank=True, null=True)
+    jboss_eap_cores_physical = models.FloatField(default=0.0, blank=True, null=True)
+    jboss_eap_cores_virtual = models.FloatField(default=0.0, blank=True, null=True)
+    jboss_eap_instances = models.PositiveIntegerField(default=0, blank=True, null=True)
+    jboss_ws_cores_physical = models.FloatField(default=0.0, blank=True, null=True)
+    jboss_ws_cores_virtual = models.FloatField(default=0.0, blank=True, null=True)
+    jboss_ws_instances = models.PositiveIntegerField(default=0, blank=True, null=True)
+    missing_cpu_core_count = models.PositiveIntegerField(
+        default=0, blank=True, null=True
+    )
+    missing_cpu_socket_count = models.PositiveIntegerField(
+        default=0, blank=True, null=True
+    )
+    missing_name = models.PositiveIntegerField(default=0, blank=True, null=True)
+    missing_pem_files = models.PositiveIntegerField(default=0, blank=True, null=True)
+    missing_system_creation_date = models.PositiveIntegerField(
+        default=0, blank=True, null=True
+    )
+    missing_system_purpose = models.PositiveIntegerField(
+        default=0, blank=True, null=True
+    )
+    openshift_cores = models.PositiveIntegerField(default=0, blank=True, null=True)
+    openshift_operators_by_name = models.JSONField(null=False, default=dict)
+    openshift_operators_by_kind = models.JSONField(null=False, default=dict)
+    os_by_name_and_version = models.JSONField(null=False, default=dict)
+    socket_pairs = models.PositiveIntegerField(default=0, blank=True, null=True)
+    system_creation_date_average = models.DateField(blank=True, null=True)
+    vmware_hosts = models.PositiveIntegerField(default=0, blank=True, null=True)
+    vmware_vm_to_host_ratio = models.FloatField(default=0, blank=True, null=True)
+    vmware_vms = models.PositiveIntegerField(default=0, blank=True, null=True)
 
     # Note: The following attributes come exclusively from raw facts.
     # For now, that's our only option, and these are required outputs.
     # TODO Refactor how we populate these when we restructure the underlying data.
-    openshift_cluster_instances: int = 0
-    openshift_node_instances: int = 0
+    openshift_cluster_instances = models.PositiveIntegerField(
+        default=0, blank=True, null=True
+    )
+    openshift_node_instances = models.PositiveIntegerField(
+        default=0, blank=True, null=True
+    )
 
 
 def reformat_aggregate_report_to_dict(aggregated: AggregateReport) -> dict:
```

Comment:
Shouldn't we use proper DRF serializers? I get having this when we are not using proper Django models, but now this seems completely unnecessary.

They can handle nested fields. We did that for the insights report, for instance. I made a suggestion on another comment.